src/core/src/core_logic/ExecutionConfig.py

```diff
@@ -108,9 +108,9 @@ def __get_max_patch_publish_date(self, health_store_id):
         """ Obtains implicit date ceiling for published date - converts pub_off_sku_2024.04.01 to 20240401T000000Z """
         max_patch_publish_date = str()
         if health_store_id is not None and health_store_id != "":
-            split = health_store_id.split("_")
-            if len(split) == 4 and len(split[3]) == 10:
-                max_patch_publish_date = "{0}T000000Z".format(split[3].replace(".", ""))
+            split = health_store_id.rsplit("_", 1)
+            if len(split) == 2 and len(split[1]) == 10:
+                max_patch_publish_date = "{0}T000000Z".format(split[1].replace(".", ""))
```

Comment:
Can you please write a debug log if it reaches this far and the conditions to convert successfully aren't met. e.g. split happens and then the data doesn't look right.

Reply:
In this scenario, line 115 would log -> [EC] Getting max patch publish date. [MaxPatchPublishDate=][HealthStoreId=pub_off_sku_2020.9.29]

Reply:
Sounds good - closing.